theano/scalar/basic.py

```diff
@@ -851,6 +851,8 @@ class ScalarOp(Op):
     nin = -1
     nout = 1
 
+    __props__ = ("output_types_preference", "name")
```

Comment:
In an ideal case, yes. But this class have subclass that need different \__props__. So you should remove this one or add the props to all the subclass that need it. InRange is such a subclass, maybe there is others.